cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/DynamicInvokeResolver.kt

```diff
@@ -112,7 +113,19 @@ class DynamicInvokeResolver(ctx: TranslationContext) : ComponentPass(ctx) {
         // rid of FunctionPointerType and only deal with FunctionTypes.
         val pointerType: FunctionPointerType =
             when (val type = expr.type) {
-                is FunctionType -> type.pointer() as FunctionPointerType
+                is FunctionType -> {
+                    when (val pointerType = type.pointer()) {
+                        is FunctionPointerType -> pointerType
+                        is ProblemType -> {
+                            // TODO: Can we handle this case somehow?
```

Comment:
I would assume that we should just log a warning in this case and return/ignore

Reply:
Done